Formula/sile.rb

```diff
@@ -12,20 +12,49 @@ class Sile < Formula
     depends_on "libtool" => :build
   end
 
-  bottle :disable, "LuaRocks requirements preclude bottling"
-
   depends_on "pkg-config" => :build
   depends_on "harfbuzz"
   depends_on "fontconfig"
   depends_on "libpng"
   depends_on "lua"
   depends_on "icu4c"
 
-  depends_on "lpeg" => :lua
-  depends_on "luaexpat" => :lua
-  depends_on "luafilesystem" => :lua
+  resource "lpeg" do
+    url "http://www.inf.puc-rio.br/~roberto/lpeg/lpeg-1.0.1.tar.gz"
+    sha256 "62d9f7a9ea3c1f215c77e0cadd8534c6ad9af0fb711c3f89188a8891c72f026b"
+  end
+
+  resource "luaexpat" do
+    url "https://matthewwild.co.uk/projects/luaexpat/luaexpat-1.3.0.tar.gz"
+    sha256 "d060397960d87b2c89cf490f330508b7def1a0677bdc120531c571609fc57dc3"
+  end
+
+  resource "luafilesystem" do
+    url "https://github.com/keplerproject/luafilesystem/archive/v_1_6_3.tar.gz"
+    sha256 "5525d2b8ec7774865629a6a29c2f94cb0f7e6787987bf54cd37e011bfb642068"
+  end
+
+  # For some reason it insists on doing another git clone at build because of
+  # the contents of the rockspec. This git clone is done over the git://
+  # protocol because life is bad. This should be reported upstream.
```

Comment:
You can use `luarocks make` to get it to use an already staged source tree.

Reply:
👍. I'd kind of wedged this PR on hold waiting for an upstream release, but I'll remember to implement that one that way if we ever get there.

Reply:
```
Josephs-MacBook-Pro:~ joe$ brew formula-analytics sile
install events in the last 30 days for sile
=====================================================================================================
1 | sile --HEAD                                                                         | 7 | 100.00%
=====================================================================================================
Total                                                                                   | 7 |    100%
=====================================================================================================
Josephs-MacBook-Pro:~ joe$ 
```
Maybe just pick a revision and call it a day?

Reply:
Just to check I'm not chasing you up the wrong tree here: You're happy for me to simply use the latest upstream commit of `sile` pending a new release to get this done?

Reply:
Yup. Every install in the last 30 days was --HEAD and 1/3 over the last 100 days. So I don't think we need to stand on ceremony at this point.

Reply:
Cool, I'll finish this one off this afternoon 👍.

Reply:
thanks :)